### ISSUE TITLE
Add pkg-config and version information to every jhbuild module

### ISSUE THREAD

**Issue by dpino** [Igalia/webkit-container-sdk#60]
Moved from https://github.com/Igalia/webkit-container-sdk/pull/51#issuecomment-2378877909

When `pkg-config` and `version` information are present in a jhbuild module, jhbuild will compare the version requested to the version of the library installed via system packages (in case the package was installed). If the system package is newer, then the module is skipped.

Avoiding building modules that are already installed as system libraries will reduce the size of the image. Having `pkg-info` and `version` present will also help to inadvertely downgrade a package that is already installed (In case you need to downgrade a package, simply remove the `pkg-config` or `version` from the module configuration).

In any case, it seems this mechanism only works in jhbuild for tarballs. I have pushed a merge-request to make the same logic work with git branches:
  - [Add support for attribute 'version' in Git branches](https://gitlab.gnome.org/GNOME/jhbuild/-/merge_requests/256)

We have many modules setup as git branches, so we need to get that merge-request merged and jhbuild updated before implementing this change cc @TingPing

**Comment by TingPing** [comment]
FWIW I don't really see the point of this. We control the versions in the SDK, sans micro version updates by Ubuntu. So we just remove any modules we don't need whenever.

**Comment by dpino** [comment]
> FWIW I don't really see the point of this. We control the versions in the SDK, sans micro version updates by Ubuntu. So we just remove any modules we don't need whenever.

The `meson` module was part of the JHBuild moduleset for a long time even though it was not necessary (https://github.com/Igalia/webkit-container-sdk/pull/37). The same is currently happening with the `openh264` module, as I explained in https://github.com/Igalia/webkit-container-sdk/pull/51#issuecomment-2378877909. 

IMHO, the best way to inadvertely downgrading a module via JHBuild, or installing a module that it's not necessary, is to attach the pkg-config information to every module (whenever possible). It may also help to save time in the build of the JHBuild modules and save space in the image.

**Comment by nikolaszimmermann** [comment]
@dpino any update here? was the upstream thing merged?

**Comment by dpino** [comment]
> [@dpino](https://github.com/dpino) any update here? was the upstream thing merged?

Yes, the JHBuild merge-request was merged long time ago https://gitlab.gnome.org/GNOME/jhbuild/-/merge_requests/256 

WebKit Container SDK always pulls the latest version of JHBuild https://github.com/Igalia/webkit-container-sdk/blob/main/images/wkdev_sdk/Containerfile#L105, so this change is already in use in WebKit Container SDK.
